### PR TITLE
[MO, TF] Support Custom Wide and Deep CTR model by MO

### PR DESCRIPTION
**Pull request by rkazants**
**Root cause analysis:** The requested Wide and Deep model contains a new operation *SparseSegmentMean*.

**Solution:** This PR contains implementation to support Wide & Deep model sub-graph with *SparseSegmentMean* operation. This support is implemented using two transformations:
1. *EmbeddingSegmentSum* fusion (for this sub-graph) transformation is extended with  *SparseSegmentMean* operation case for which this sub-graph is fused to *EmbeddingSegmentMean* (MO internal operation)
2. *EmbeddingSegmentMean* decomposition transformation generates a sub-graph with *EmbeddingSegmentSum* and a tail that computes normalization coefficients for summed-up embedding vectors of each object and normalizes
3. Remove limitation to support i64 in *ScatterND* operation, and align input types for *EmbeddingSegmentSum* operation since segments number and indices can have different integer types dictated by model specific but the specification requires the same types.
4. Correct documentation to return original i64 type

**Tickets:** 61842, 62901

Code:
* [x]  Comments
* [x]  Code style (PEP8)
* [x]  Transformation generates reshape-able IR
* [x]  Transformation preserves original framework node names

Validation:
* [x]  Unit tests - N/A
* [x]  Framework operation tests - Not possible to do with a simple layer tests. I have to program a solid part of the model. Decided to cover it through e2e model testing
* [x]  Transformation tests - Decided not do since the sub-graph is too huge, useless, and time-consuming in this case
* [x]  Model Optimizer IR Reader check

Documentation:
* [x]  Supported frameworks operations list
* [x]  Guide on how to convert the **public** model - N/A since the model only template inferred
* [x]  User guide update - N/A since the model only template inferred

### CODE REVIEW

**Comment by achetver** [comment]
In general looks good, but looks like we have failed e2e test for TF wide and Deep model with: RuntimeError: IR generation failed with 1 exit code

**Comment by rkazants** [comment]
> In general looks good, but looks like we have failed e2e test for TF wide and Deep model with: RuntimeError: IR generation failed with 1 exit code

Now I am fixing this test. I need to change input type to the original {i64}.

File: model-optimizer/extensions/front/tf/embedding_segments_mean_decomposition.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+from mo.ops.unsqueeze import Unsqueeze
+
+
+class EmbeddingSegmentsMeanDecomposition(FrontReplacementPattern):
+    """
+    This transformation decomposes EmbeddingSegmentsMean operation into EmbeddingSegmentSum operations taking into
+    account that summed up embedding vectors for each vector must be normalized appropriately by a coefficient
+    equal to a number of gathered embedding vectors for each object. If there is no gathered embedding vector
+    for an object, the coefficient equals one.
+    """
```
Comment (rkazants) [inline model-optimizer/extensions/front/tf/embedding_segments_mean_decomposition.py:29]: Add visualization (a graph before and after transformation)

**Review by achetver — APPROVED** [review]
Looks good to me

File: model-optimizer/extensions/front/tf/embedding_segments_mean_decomposition.py (hunk 1 above)
Comment (rkazants) [reply model-optimizer/extensions/front/tf/embedding_segments_mean_decomposition.py:39]: Done

**Review by mvafin — APPROVED** [review]

**Comment by rkazants** [comment]
> > In general looks good, but looks like we have failed e2e test for TF wide and Deep model with: RuntimeError: IR generation failed with 1 exit code
> 
> Now I am fixing this test. I need to change input type to the original {i64}.

Done

**Pull request merged**